### PR TITLE
Convert regr

### PR DESCRIPTION
**Pull request by dinossimpson**
Contains the regression test for SpecAxis.convert_to_unit().
Uses 3 doppler equivalencies and tests the conversion between those units

### CODE REVIEW

File: test_convert_to_unit.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+from pyspeckit.spectrum.units import SpectroscopicAxis
+import astropy.units as u
+import random
+import numpy as np
+import inspect
+import pytest
+
+def test_convert_to_unit():
+	run_with_assert = False
```
Comment (keflavich) [inline test_convert_to_unit.py]: why not make this an argument to the function?

File: test_convert_to_unit.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	for from_unit, to_unit, _, __ in u.doppler_optical(1*u.Hz):
+		sp = SpectroscopicAxis(np.linspace(-100,100,100), unit=from_unit, refX=23.2*u.Hz, velocity_convention='optical')
+		sp.convert_to_unit(to_unit)
+		print sp
+		if(run_with_assert):
+			assert sp.unit == to_unit
```
Comment (keflavich) [inline test_convert_to_unit.py]: something is wrong with the tabs here

File: test_convert_to_unit.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import numpy as np
+import inspect
+import pytest
+
+def test_convert_to_unit():
+	run_with_assert = False
+	for from_unit, to_unit, _, __ in u.doppler_optical(1*u.Hz):
+		sp = SpectroscopicAxis(np.linspace(-100,100,100), unit=from_unit, refX=23.2*u.Hz, velocity_convention='optical')
+		sp.convert_to_unit(to_unit)
+		print sp
```
Comment (keflavich) [inline test_convert_to_unit.py]: remove the print statements

File: test_convert_to_unit.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import astropy.units as u
+import random
+import numpy as np
+import inspect
+import pytest
+
+def test_convert_to_unit():
+	run_with_assert = False
+	for from_unit, to_unit, _, __ in u.doppler_optical(1*u.Hz):
+		sp = SpectroscopicAxis(np.linspace(-100,100,100), unit=from_unit, refX=23.2*u.Hz, velocity_convention='optical')
```
Comment (keflavich) [inline test_convert_to_unit.py]: could you wrap this at 80 characters?

File: test_convert_to_unit.py (hunk 2 above)
Comment (keflavich) [inline test_convert_to_unit.py]: actually, everywhere: use 4 spaces, not 8 spaces, per indent

Comment (dinossimpson) [inline test_convert_to_unit.py]: sth was wrong, it didn't look like that locally

**Pull request merged**